packages/datadog_session_replay/lib/src/capture/element_recorders/common_nodes.dart

```diff
@@ -0,0 +1,250 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-Present Datadog, Inc.
+
+import 'package:flutter/material.dart';
+
+import '../../extensions.dart';
+import '../../sr_data_models.dart';
+import '../capture_node.dart';
+
+@immutable
+class CapturedBorderStyle {
+  final double? cornerRadius;
+  final double? width;
+  final String? color;
+
+  const CapturedBorderStyle({
+    required this.cornerRadius,
+    required this.width,
+    required this.color,
+  });
+
+  static CapturedBorderStyle? fromShapeBorder(
+    ShapeBorder? shape,
+    CapturedViewAttributes attributes,
+  ) {
+    switch (shape) {
+      case final StadiumBorder _:
+        final shortSide = attributes.paintBounds.shortestSide;
+        return CapturedBorderStyle(
+          cornerRadius: shortSide / 2,
+          width: shape.side.width,
+          color: shape.side.color.toHexString(),
+        );
+      case final CircleBorder _:
+        final shortSide = attributes.paintBounds.shortestSide;
+        return CapturedBorderStyle(
+          cornerRadius: shortSide / 2,
+          width: shape.side.width,
+          color: shape.side.color.toHexString(),
+        );
+      case final RoundedRectangleBorder shape:
+        // TODO: TextDirection
```

Comment:
suggestion: it is better also to add JIRA ticket number for all TODO items

Reply:
Actually removing this TODO as it doesn't matter unless we support per-corner radius